### PR TITLE
Add `libsafec` packages to Mariner

### PR DESCRIPTION
**Pull request by bossmc**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Security requirements sometimes demand use of Annex K secure library functions (`strcpy_s` and friends).  These are not available in `glibc` so this package backfills them with a `libsafec` library that provides the wrapped versions of the functions with additional safety checks.

###### Change Log  
- Add Annex K secure library functions in `libsafec` to Mariner

###### Does this affect the toolchain?  
No

###### Associated issues  
None

###### Links to CVEs  
None

###### Test Methodology
Local build

### CODE REVIEW

File: SPECS/safeclib/safeclib.signatures.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+ "Signatures": {
```
Comment (jslobodzian) [inline SPECS/safeclib/safeclib.signatures.json:2]: Also need to update the cgmanifest file, the SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md file and the SPECS/LICENSES-AND-NOTICES/data/licenses.json file

For the latter two file, it looks like this is an original SPEC file for Mariner, so the SPEC file would be categorized with the Microsoft MIT license.

Comment (bossmc) [reply SPECS/safeclib/safeclib.signatures.json:2]: Done

**Review by jslobodzian — APPROVED** [review]

**Comment by bossmc** [comment]
I've fixed the cgmanifest error (@PawelWMS helpfully pointed me at the `update_cgmanifest.py` tool) and I've uploaded the source package too, so I think this is only waiting on a click of the big green button from someone with the appropriate power.

Recurring issue that the spec linter wants to delete my changelog entry, but I'm ignoring that.

File: SPECS/safeclib/safeclib.spec [hunk 2]
```diff
@@ -0,0 +1,72 @@
+Summary:        C11 Annex K functions
+Name:           libsafec
+Version:        3.7.1
+Release:        1%{?dist}
+License:        MIT
+URL:            https://github.com/rurban/safeclib
+Source0:        https://github.com/rurban/safeclib/releases/download/v3.7.1/safeclib-%{version}.tar.xz
```
Comment (PawelWMS) [inline SPECS/safeclib/safeclib.spec:7]:
```suggestion
Source0:        https://github.com/rurban/safeclib/releases/download/v%{version}/safeclib-%{version}.tar.xz
```

File: SPECS/safeclib/safeclib.spec [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+Requires:       libsafec = %{version}-%{release}
+
+%description -n libsafec-devel
+Development files for libsafec
+
+%package -n libsafec-check
+Summary:        Tools to detect use of unsafe libc APIs
+
+%description -n libsafec-check
+Tools to detect use of unsafe libc APIs
```
Comment (PawelWMS) [inline SPECS/safeclib/safeclib.spec:30]: The `-n` is not needed in this case and we tend to only add it, if the package name's prefix is different from the value of the `Name` tag:
```suggestion
%package devel
Summary:        Development packages for libsafec
Requires:       %{name} = %{version}-%{release}

%description devel
Development files for libsafec

%package check
Summary:        Tools to detect use of unsafe libc APIs

%description check
Tools to detect use of unsafe libc APIs
```

File: SPECS/safeclib/safeclib.spec [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+%build
+autoreconf -Wall --install
+%configure --disable-static --enable-debug --enable-strmax=0x8000
+%make_build
+
+%if %{with_check}
+%check
+export TZ=Europe/London
+make check %{?_smp_mflags}
+%endif
```
Comment (PawelWMS) [inline SPECS/safeclib/safeclib.spec:44]: No need to cut out the `%check` section for non-test builds. This section is automatically skipped anyway.
```suggestion
%check
export TZ=Europe/London
make check %{?_smp_mflags}
```

File: SPECS/safeclib/safeclib.spec [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+%files -n libsafec-devel
+%license COPYING
+%{_libdir}/libsafec.so
+%{_includedir}/safeclib
+%{_mandir}/man3/*
+
+%files -n libsafec-check
+%license COPYING
+%{_bindir}/check_for_unsafe_apis
+%{_mandir}/man1/*
```
Comment (PawelWMS) [inline SPECS/safeclib/safeclib.spec:67]: Same thing for the `-n` switch in the `%files` sections:

```suggestion
%files
%license COPYING
%{_libdir}/libsafec.so.*

%files devel
%license COPYING
%{_libdir}/libsafec.so
%{_includedir}/safeclib
%{_mandir}/man3/*

%files check
%license COPYING
%{_bindir}/check_for_unsafe_apis
%{_mandir}/man1/*
```

File: SPECS/LICENSES-AND-NOTICES/data/licenses.json [hunk 6]
```diff
@@ -2048,6 +2048,7 @@
                 "rubygem-semantic_puppet",
                 "rubygem-test-unit",
                 "rubygem-thor",
+                "safeclib",
```
Comment (PawelWMS) [inline SPECS/LICENSES-AND-NOTICES/data/licenses.json:2051]: Still needs to be changed to the value of the `Name` tag from the spec.

File: SPECS/safeclib/safeclib.spec [hunk 7]
```diff
@@ -0,0 +1,72 @@
+Summary:        C11 Annex K functions
+Name:           libsafec
```
Comment (PawelWMS) [inline SPECS/safeclib/safeclib.spec:2]: Please always use the same value for the spec, its directory, and the `Name` tag inside the spec. Only exception is when one directory contains multiple spec files like `SPECS/coredns/coredns-[versionX].spec` for instance.

**Review by PawelWMS — COMMENTED** [review]
Sorry for the late review, @bossmc! I just found a few minutes to take a look.

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**